### ISSUE TITLE
Regression for typing.Annotated when aliased

### ISSUE THREAD

**Issue by fellhorn** [microsoft/pyright#9092]
**Describe the bug**

`Pyright` 1.1.379 seems to have introduced a different behavior for `typing.Annotated` type forms when:

* `enableExperimentalFeatures: true` is set
* `typing.Annotated` is aliased in another module

I assume this is due to the provisional support for draft PEP 747 and the `TypeForm` special form.

We discovered this because it breaks [jaxtyping](https://github.com/patrick-kidger/jaxtyping) type annotations, which internally aliases `typing.Annotated`.

**Code or Screenshots**

The file to type check:

```python
from typing import Annotated

# These two alias typing.Annotated
from .indirection import IndirectAnnotated
from jaxtyping import Int32


def test(
    inp1: Annotated[int, "metadata"],  # This is valid
    inp2: IndirectAnnotated[int, "metadata"], # "metadata" is not defined (reportUndefinedVariable)
    inp3: Int32[int, "metadata"], # "metadata" is not defined (reportUndefinedVariable)
):
    pass
```

`indirection.py`

```python
from typing import (
    Annotated as IndirectAnnotated,
)

__all__ = [
    "IndirectAnnotated",
]
```

`pyrightconfig.json`

```json
{
    "include": [
        "./*",
    ],
    "enableExperimentalFeatures": true,
}
```

> If your code relies on symbols that are imported from a third-party library, include the associated import statements and specify which versions of those libraries you have installed.

`indirection.py` provides a minimal example for `jaxtyping`'s internal logic.

**VS Code extension or command-line**

All `pyright` versions `>=1.1.379` as well as `pylance` `v2024.9.101 (pre release)` produce the issue:

```sh
> pyright -p pyrightconfig.json 
/workspaces/pyright_test/test.py
  /workspaces/pyright_test/test.py:11:35 - error: "metadata" is not defined (reportUndefinedVariable)
  /workspaces/pyright_test/test.py:12:22 - error: "metadata" is not defined (reportUndefinedVariable)
2 errors, 0 warnings, 0 informations 
```

**Comment by fellhorn** [comment]
I found a smaller example to reproduce the issue for the pyright playground:

Code sample in [pyright playground](https://pyright-play.net/?pyrightVersion=1.1.382&pythonVersion=3.12&enableExperimentalFeatures=true&code=GYJw9gtgBALgngBwJYDsDmUkQWEMoCCKKYMAhjAKYAmAULQJIrVIiUDGMRJ5V1UAXkLFSFGvWqVgsSgGcYAClpQVmFAgCMALmE8x1ANqoYAGigAiCJXLUKZcwF0zUAMRQAKgAskszL4BuZAA2SHSqaggATDpMLGyc3KJ8RiimFlY2do5mbpbWZLbk5n5QPFCSwKg0UApsOHgAqsxSVdQAamQgSGQARkGUAJS0A1rKqghksrJAA)

```python
from typing import Annotated

IndirectAnnotated = Annotated

def test(
    inp1: Annotated[int, "metadata"],  # This is valid
    inp2: IndirectAnnotated[int, "metadata"], # "metadata" is not defined (reportUndefinedVariable)
):
    pass
```

**Comment by erictraut** [comment]
Thanks for helping shake out the bugs in the provisional implementation of `TypeForm`.

This will be addressed in the next release.

**Comment by fellhorn** [comment]
Thanks a lot, Eric! It's always amazing to see how quickly you react to bug reports here in `pyright`.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.383.
